invenio_records_rest/__init__.py

```diff
@@ -22,7 +22,634 @@
+
+Searching
+---------
+Invenio-Records-REST offers a rich set of fundamental search operations using
+query strings, filters, sorting and pagination and also more advanced
+operations including aggregations and suggestions.
+
+``/records/?q=&size=10&page=1&sort=bestmatch&type=test``
+
+Faceted Search
+~~~~~~~~~~~~~~
+
+Faceted search is enabled through filters and aggregations. It can be used to
+obtain subsets of documents which meet certain criteria, group results by
+categories, and provide an easy to navigate faceted menu.
+
+``/records/?type=test&type=anoterhvalye/``
+
+
+Aggregations
+++++++++++++
+By exposing the Elasticsearch API by default, we can use the advanced
+features it provides, such as the `aggregations framework
+<https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations.html>`_.
+
+These features include:
+
+- `bucketing <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-bucket.html>`_
+  "used to group the documents by a certain criterion".
+- `metric <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-metrics.html>`_
+  "used to create different types of metrics from values extracted from the documents
+  being aggregated".
+- `matrix <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-matrix.html>`_
+  "used to produce a matrix result based on the values extracted from the requested document fields".
+- `pipeline <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-pipeline.html>`_
+  "used to aggregate the output of other aggregations and their associated metrics".
+
+
+Filters
++++++++
+Filters can be applied at different points of a query having different effects
+for aggregations, but not for search queries.
+
+
+To set the aggregations and filters you want you can modify the
+`RECORDS_REST_FACETS`.
+
+>>> from .facets import terms_filter
+>>> app.config['RECORDS_REST_FACETS'] = {
+...     'index_name': {
+...         'aggs': {
+...             'type': {'terms': {'field': 'type'}}
+...         },
+...         'post_filters': {
+...             'type': terms_filter('type'),
+...         },
+...         'filters': {
+...             'typefilter': terms_filter('type'),
+...         }
+...     }
+... }
+
+Sorting
+~~~~~~~
+Sorting is based by default on a relevance score, but this can
+be configured as well. The following ways are possible:
+
+- `Sorting by field values <https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html#_sorting_by_field_values>`_
+- `Multilevel sorting <https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html#_multilevel_sorting>`_
+- `Sorting on multilevel fields <https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html#_sorting_on_multivalue_fields>`_
+
+This can be configured through the `RECORDS_REST_SORT_OPTIONS`:
+
+>>> RECORDS_REST_SORT_OPTIONS = dict(
```

Comment:
Maybe it's worth mentioning here about how the descending sorting is translated in the querystring via a minus (`-`) sign: Bestmatch descending -> `-bestmatch`